### DISCUSSION TITLE
Multiple queries with the same object as result?

### DISCUSSION THREAD

**Discussion by wuifdesign** [TanStack/query#566] [General]
Hi,

I have a question about best practice with this library. i have multiple queries returning the same object lets say "user"

- /users-with-books
- /friends
- /friends-by-user/{id}
- /user/{id}

when running the application i now have the same user object about 100 times stored in react query.

now my question: when i update the user, what is the best way to update all 100 calls using the same user. fireing 100 calls to the backend dosn't seam like a good idea. 

When Using redux i just would update my single user in the store, as i would normalize all of the calls and it would update anywhere, but i don't know how to handle this with react query. is there a way to link results, or some way to normalize them?

thx

**Comment by CreativeTechGuy** [comment]
This looks to partially overlap with #575 in the case where `/user/{id}` can reuse the same data from another request.

My suspicion is that this library was designed with lots of redundant network requests in mind. That explains the instant cache invalidation and constant refreshes.

**Reply by wuifdesign** [reply]
no, the problem of #575 is already solved in the docs. even with the same example: https://github.com/tannerlinsley/react-query#initial-data-from-cache

my problem is, i have more than 2 requests and i want to update data.

**Comment by wuifdesign** [comment]
any updates on this?

**Reply by ms88privat** [reply]
I think you should go with option 4 and have a look at this answer: https://github.com/tannerlinsley/react-query/issues/521#issuecomment-637271629

So you only fetch, what is currently needed and everything else will be fetched at a later time anyway.
